iterator.py

```diff
@@ -1,19 +1,20 @@
-import os, re, requests
+import os 
+import re 
+import requests
 rootdir = '_posts'
 
 for subdir, dirs, files in os.walk(rootdir):
     for file in files:
         filename = os.path.join(subdir, file)
 
-    	f = open(filename, "r")
-    	contents = f.readlines()
-    	f.close()
+        contents = ""
```

Comment:
Using the `with ... as ...:` syntax, we ensure that files are properly closed, and memory is properly managed.